src/schema-walker.ts

```diff
@@ -28,9 +28,11 @@ export type SchemaHandler<
   Variant extends HandlingVariant = "regular",
 > = (params: SchemaHandlingProps<T, U, Context, Variant>) => U;
 
+export type CustomBrand = string | symbol;
+
 export type HandlingRules<U, Context extends FlatObject = {}> = Partial<
   Record<
-    z.ZodFirstPartyTypeKind | ProprietaryBrand,
+    z.ZodFirstPartyTypeKind | ProprietaryBrand | CustomBrand,
     SchemaHandler<any, U, Context> // keeping "any" here in order to avoid excessive complexity
```

Comment:
`any` should be replaced

Reply:
addressed, but `any` moved to another place intentionally, so that it should be overridden by actual implementation